scripts/setup-proof.py

```diff
@@ -22,7 +22,7 @@ def read_proof_template(filename):
 
 def write_proof_template(lines, filename, directory):
     with open(os.path.join(directory, filename), "w") as data:
-        data.write('\n'.join(lines))
+        data.write('\n'.join(lines)+'\n')
```

Comment:
Probably not a huge issue if we have a small number of lines, but this could be made more efficient. Instead of constructing a very large string using `join`, we could do:

```py
        data.writelines("{}\n".format(line) for line in lines)
```

This would just iterate over a generator object.

Reply:
Done.